### ISSUE TITLE
[platform_interface] Follow-up to plugin platform interface work

### ISSUE THREAD

**Issue by navaronbracke** [juliansteenbakker/mobile_scanner#968]
Due to the requested priority of #916 (including all of its bugfixes) I've opted to prepare an initial version 5.0.0 beta release.

However, the current beta version still lacks a few things.

- using `extension types` for our JS bindings on the web
- relanding support for JSQR (if needed)
- adding integration / unit tests to `mobile_scanner`
- WASM support for the web, since Flutter is about to enable that for its users

Regarding tests, PR's are welcome to add some of those.
I will be taking on the extension types / WASM support work myself.

**Comment by juliansteenbakker** [comment]
@navaronbracke are there any other items that should be fixed from this list after #1402 is merged? Or is it safe to release 7.0.0, and move these items to 7.1 or 8.0?

**Comment by navaronbracke** [comment]
No, I don't see anything else that would block that release. I'd keep the list above around for tracking the other "nice to have"s, though. (some of which is fine for a 7.1)
